src/Ushahidi/Modules/V5/Actions/Permissions/Queries/FetchPermissionsQuery.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Actions\Permissions\Queries;
+
+use App\Bus\Query\Query;
+
+class FetchPermissionsQuery implements Query
+{
+    const DEFAULT_LIMIT = 0;
+    const DEFAULT_ORDER = "ASC";
+    const DEFAULT_SORT_BY = "id";
+    const AVAILABLE_SEARCH_FIELDS = ['q' /* LIKE name */];
```

Comment:
If there's more than just one field maybe it's worth creating a DTO class to represent fields we accept?  Something like

```php
class PermissionsAvailableSearchFieldsDTO
{
    private $query;

    public function __construct(string $query)
    {
         $this->query = $query;
    }

    public function getQuery(): string
    {
        return $this->query;
    }
}
```

If it's just query/q then maybe we should just convert it to the string? There would be less operation around it to get that `q` value.